### ISSUE TITLE
useButton should not return `ref`

### ISSUE THREAD

**Issue by Andarist** [adobe/react-spectrum#1383]
# 🐛 Bug Report 

This is rather surprising - most of the hooks only accept ref and don't actually return it. It IMHO makes great sense to only accept refs (better composability etc) and ideally consumer should not get any ref back, from what I believe this ref is not meant to be returned and it's only meant to be synchronized internally, here:
https://github.com/adobe/react-spectrum/blob/1d9905e7c8308cfbb2a1e03bfc22b2dce89cbb12/packages/%40react-aria/focus/src/useFocusable.tsx#L39

## 🤔 Expected Behavior

no `ref` in the return value of `useButton`

## 😯 Current Behavior

`ref` in the return value of `useButton`

**Comment by ktabors** [comment]
It is something we'd need to look into, I'm not sure why this difference exists. I'm hesitant about changing it without testing and this could be a breaking change for projects that are using that returned ref.

**Comment by Andarist** [comment]
Just a note - the issue is about `useButton` and not `useFosucable` (I've made a mistake while reporting it). The mention of `useFocusable` is still somewhat relevant as it's the part that actually is using this ref internally as `useButton` builds on top of `useFocusable`.

I understand the breaking changes concerns - it IMHO is a clear(-ish) mistake that this has been returned from there as no other hooks do that and if this would be a deliberate decision then probably most hooks would also return it. However, you need to make a decision if it's fine to change this at this point in time - maybe it's something that will have for v4 release?

**Comment by ktabors** [comment]
Yes I also traced that various things pass the useFocusable ref back to the dom which is different than most of the other react-aria handling of refs. We really want to research this and figure out if there is a reason.

**Comment by devongovett** [comment]
I don't think `useButton` is returning a ref currently. Looks like it was removed a long time ago: a21308574443d2b037e71616c7f7a0208e8338a2
